libcni/conf.go

```diff
@@ -189,10 +195,10 @@ func LoadConfList(dir, name string) (*NetworkConfigList, error) {
 		// A little extra logic so the error makes sense
 		switch {
 		// neither configlists nor config files found
-		case len(files) == 0 && err == NoConfigsFoundError:
+		case len(files) == 0 && reflect.TypeOf(err).Name() == "NoConfigsFoundError":
 			return nil, err
 		// config lists found but no config files found
-		case len(files) != 0 && err == NoConfigsFoundError:
+		case len(files) != 0 && reflect.TypeOf(err).Name() == "NoConfigsFoundError":
```

Comment:
The Go convention here is a type assertion to check for an error type instead of reflect.  [See here](https://blog.golang.org/error-handling-and-go), specifically the part about "type assertion".